app/src/org/commcare/activities/UpdateActivity.java

```diff
@@ -315,13 +323,12 @@ protected void deliverError(UpdateActivity receiver,
 
     @Override
     public CustomProgressDialog generateProgressDialog(int taskId) {
-        if (taskId != DIALOG_UPGRADE_INSTALL) {
+        if (CommCareApplication._().isConsumerApp()) {
+            return ConsumerAppsUtil.getGenericConsumerAppsProgressDialog(taskId, false);
+        } else if (taskId != DIALOG_UPGRADE_INSTALL) {
             Log.w(TAG, "taskId passed to generateProgressDialog does not match "
                     + "any valid possibilities in CommCareSetupActivity");
             return null;
-        }
-        if (CommCareApplication._().isConsumerApp()) {
-            return ConsumerAppsUtil.getGenericConsumerAppsProgressDialog(taskId, false);
         } else {
             return generateNormalUpdateDialog(taskId);
```

Comment:
Would be nice if this said `UpdateInstall` or `Install` instead of `Update`, because this dialog is only shown when installing an update, not downloading it.